### PR TITLE
Adds flag to throw for invalid addon locations

### PR DESCRIPTION
**Pull request by tmquinn**
This also surfaces the warning outside of the logger

### CODE REVIEW

File: lib/models/package-info-cache/package-info.js [hunk 1]
```diff
@@ -346,21 +347,26 @@ class PackageInfo {
...
+        msg = `${msg}\n  specifies invalid, malformed or missing addons at relative paths`;
         invalidPackages.forEach(packageInfo => {
           let relativePath = path.relative(this.realPath, packageInfo.realPath);
-          logger.info(`    '${relativePath}'`);
+          msg = `${msg}\n    '${relativePath}'`;
         });
       }
+
+      const config = require('../../utilities/get-config')();
+      if (config.get('errorOnInvalidAddon')) {
```
Comment (rwjblue) [inline lib/models/package-info-cache/package-info.js]: Why is this a flag and not the default? What circumstances are we intentionally wanting to swallow invalid addon paths for?

Comment (tmquinn) [reply lib/models/package-info-cache/package-info.js]: I know, right? This has bitten us many times, I think the reason for the flag is that the behavior is radically different from the current behavior. Previously it didn't even throw a warning without the DEBUG flag set...

Comment (rwjblue) [reply lib/models/package-info-cache/package-info.js]: So the reason it is behind a flag is that it is a potentially breaking change?

Comment (tmquinn) [reply lib/models/package-info-cache/package-info.js]: Ya, per @stefanpenner

Comment (rwjblue) [reply lib/models/package-info-cache/package-info.js]: I think we should:

* have it always emit a warning to the console (this is the only way to get these cleaned up throughout the ecosystem)
* have it throw an error when a `process.env` variable is set

Comment (tmquinn) [reply lib/models/package-info-cache/package-info.js]: OK, so this PR does the first one already, so I will just switch the flag from .ember-cli to an env...

File: lib/models/package-info-cache/package-info.js [hunk 2]
```diff
@@ -346,21 +347,26 @@ class PackageInfo {
...
+        throw msg;
+      } else {
+        console.warn(chalk.yellow(msg));
```
Comment (rwjblue) [inline lib/models/package-info-cache/package-info.js]: Do we not have access to a console-ui instance here? In general, we should try not to use `console` if we can help it 🤔

**Comment by tmquinn** [comment]
I looked around for it... it's not on this in this context...

File: lib/models/package-info-cache/package-info.js (hunk 2 above)
Comment (rwjblue) [reply lib/models/package-info-cache/package-info.js]: We (at least) sometimes have `this.project` which has `this.project.ui` on it.

File: lib/models/package-info-cache/package-info.js [hunk 3]
```diff
@@ -346,21 +346,25 @@ class PackageInfo {
...
-        logger.info('  specifies invalid, malformed or missing addons at relative paths');
+        msg = `${msg}\n  specifies invalid, malformed or missing addons at relative paths`;
         invalidPackages.forEach(packageInfo => {
           let relativePath = path.relative(this.realPath, packageInfo.realPath);
-          logger.info(`    '${relativePath}'`);
+          msg = `${msg}\n    '${relativePath}'`;
         });
       }
+
+      if (process.env.ERROR_ON_INVALID_ADDON) {
```
Comment (rwjblue) [inline lib/models/package-info-cache/package-info.js]: Let's prefix this with `EMBER_CLI_`:

```suggestion
      if (process.env.EMBER_CLI_ERROR_ON_INVALID_ADDON) {
```

**Comment by rwjblue** [comment]
Thank you!

**Pull request merged**